src/AzureAppConfigurationImpl.ts

```diff
@@ -157,22 +174,51 @@ export class AzureAppConfigurationImpl extends Map<string, any> implements Azure
             keyValues.push([key, value]);
         }
 
-        this.clear(); // clear existing key-values in case of configuration setting deletion
+        this.#configMap.clear(); // clear existing key-values in case of configuration setting deletion
         for (const [k, v] of keyValues) {
-            this.set(k, v);
+            this.#configMap.set(k, v);
         }
+        // also construct hierarchical data object from map
+        this.#constructDataFromMap();
     }
 
     /**
      * Load the configuration store for the first time.
      */
     async load() {
         await this.#loadSelectedAndWatchedKeyValues();
-
         // Mark all settings have loaded at startup.
         this.#isInitialLoadCompleted = true;
     }
 
+    /**
+     * Construct hierarchical data object from map.
+     *
+     * @remarks
+     * This method is used to construct hierarchical data object from map, and it is called after the initial load and each refresh.
+     */
+    #constructDataFromMap() {
+        // clean data object in-place
+        for (const key of Object.keys(this.data)) {
+            delete this.data[key];
+        }
+        // construct hierarchical data object from map
+        for (const [key, value] of this.#configMap) {
+            const segments = key.split(this.#SEP);

```

Comment:
Should we handle the corner case when key = "a...b"

Reply:
Good catch. In the case segments can be empty strings, which I think we should skip the construction for this key value. Or maybe honor the mentioned option (specifying behavior when ambiguity occurs) to silently skip or to throw error directly?

Reply:
I vote for throwing error.

Reply:
impl: throw an error/ingore depending on value of  `options.onError`.